### PR TITLE
Void networking + REI Integration (through Void)

### PR DESCRIPTION
**Pull request by Rattlyy**
tested

### CODE REVIEW

File: server/src/main/kotlin/net/horizonsend/ion/server/features/customItems/CustomItems.kt [hunk 1]
```diff
@@ -25,7 +25,10 @@ import kotlin.math.roundToInt
 // budget minecraft registry lmao
 object CustomItems {
 	// If we want to be extra fancy we can replace this with some fastutils thing later.
-	private val customItems: MutableMap<String, CustomItem> = mutableMapOf()
+	val all
+		get() = _customItems.toMap() // not mutable
+
+	private val _customItems: MutableMap<String, CustomItem> = mutableMapOf()
```
Comment (Peter-Crawley) [inline server/src/main/kotlin/net/horizonsend/ion/server/features/customItems/CustomItems.kt:31]:
```suggestion
	private val customItems: MutableMap<String, CustomItem> = mutableMapOf()
	val all get() = customItems.toMap()
```

File: server/src/main/kotlin/net/horizonsend/ion/server/features/client/VoidNetwork.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package net.horizonsend.ion.server.features.client
+
+import net.starlegacy.util.Tasks
+import org.bukkit.event.EventHandler
+import org.bukkit.event.Listener
+import org.bukkit.event.player.PlayerJoinEvent
+import org.bukkit.event.player.PlayerQuitEvent
+import java.util.*
```
Comment (Peter-Crawley) [inline server/src/main/kotlin/net/horizonsend/ion/server/features/client/VoidNetwork.kt:8]: No wildcards

File: server/src/main/kotlin/net/horizonsend/ion/server/features/client/VoidNetwork.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+class VoidNetwork : Listener {
+	@EventHandler
+	fun handshakeUser(event: PlayerJoinEvent) {
+		Tasks.syncDelay(10) {
```
Comment (Peter-Crawley) [inline server/src/main/kotlin/net/horizonsend/ion/server/features/client/VoidNetwork.kt:13]: Why delay?

File: server/src/main/kotlin/net/horizonsend/ion/server/features/client/Packets.kt [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+			VoidNetwork.modUsers.add(it.uniqueId)
+			PLAYER_ADD.broadcast { writeUUID(it.uniqueId) }
+
+			for (id in VoidNetwork.modUsers - it.uniqueId) {
+				PLAYER_ADD.send(it) { writeUUID(it.uniqueId) }
+			}
+		},
+		s2c = {
+			val items =
+				(
```
Comment (Peter-Crawley) [inline server/src/main/kotlin/net/horizonsend/ion/server/features/client/Packets.kt:29]: what unholy mess of formatting happened here

**Review by Peter-Crawley — CHANGES_REQUESTED** [review]

File: server/src/main/kotlin/net/horizonsend/ion/server/features/client/VoidNetwork.kt (hunk 3 above)
Comment (Rattlyy) [reply server/src/main/kotlin/net/horizonsend/ion/server/features/client/VoidNetwork.kt:13]: We need to give the user some time to finish everything else it causes problems

File: server/src/main/kotlin/net/horizonsend/ion/server/features/client/Packets.kt (hunk 4 above)
Comment (Rattlyy) [reply server/src/main/kotlin/net/horizonsend/ion/server/features/client/Packets.kt:29]: idk but it works

File: server/src/main/kotlin/net/horizonsend/ion/server/features/client/VoidNetwork.kt (hunk 3 above)
Comment (Peter-Crawley) [reply server/src/main/kotlin/net/horizonsend/ion/server/features/client/VoidNetwork.kt:13]: this isnt a reliable fix though

Comment (Rattlyy) [reply server/src/main/kotlin/net/horizonsend/ion/server/features/client/VoidNetwork.kt:13]: plus i fixed the rei thingy that broke this so technically I can remove it but if it works

Comment (Peter-Crawley) [reply server/src/main/kotlin/net/horizonsend/ion/server/features/client/VoidNetwork.kt:13]: any "fix" that relies on waiting an arbitrary amount of time is not a fix, its a hack

**Comment by Peter-Crawley** [comment]
Mrege conflicts

**Pull request closed without merging**